organizations/backends.py

```diff
@@ -34,11 +50,37 @@ def authenticate(self, *args, **kwargs):
         # superuser can log into any microsite
         site = get_current_site()
         is_default_site = site.id == settings.SITE_ID
+
         if not is_default_site and is_site_configuration_enabled() and user and not user.is_superuser:
             user_organizations = set(user.organizations.values_list('short_name', flat=True))
-            site_organization = get_value('course_org_filter')
-            if site_organization in user_organizations:
-                return user
+            site_organizations = get_value('course_org_filter')
+
+            # Site must have a defined course org.  Treat a blank value as None
+            if site_organizations is None or len(site_organizations) == 0:
+                return None
+
+            # get a sequence and remove dupes
+            if not isinstance(site_organizations, (tuple, collections.MutableSequence, type(None))):
+                site_organizations = (site_organizations, )
+            site_organizations = [org for org in set(site_organizations) if len(org)]
+
+            try:
```

Comment:
Why use exception handling and an assert here instead of

```
if not len(site_organizations) == 1:
    logger....
    return None
```

Also, this appears to be an error case due to some kind of misconfiguration. Shouldn't the logger be at least error level?